pkgconfig: Do not escape custom variables

xclaesse opened the pull request:
We need to escape space in variables that gets into cflags or libs
because otherwise we cannot split compiler args when paths contains
spaces. But custom variables are unlikely to be path that gets used in
cflags/libs, and escaping them cause regression in GStreamer that use
space as separator in a list variable.

xclaesse commented:
CC @nirbheek who did that change in https://github.com/mesonbuild/meson/pull/2611.

xclaesse commented:
For reference, here is GStreamer bug report: https://gitlab.freedesktop.org/gstreamer/gst-plugins-base/-/issues/884.

tp-m commented:
Question is if this will then break other users that put directory/file paths into custom variables and expect those to be escaped? (Who does the unescaping then?)

We seem to have paths in custom variables too, but they're ultimately prefix-based
```
toolsdir=${exec_prefix}/bin
pluginsdir=${prefix}/lib/x86_64-linux-gnu/gstreamer-1.0
pluginscannerdir=${libexecdir}/gstreamer-1.0
```
(in our case anyway)

xclaesse commented:
Spaces in variables is only a problem when you want to split them on space using `shlex.split()`. For example if you do:
```
mypath=/home/Xavier Claessens
Cflags: -I${mypath}/include -pthread
```

`pkg-config --cflags` would print `-I/home/Xavier Claessens/include -pthread`, and meson would then use `shlex.split()` on that and it gives `['-I/home/Xavier', 'Claessens/include', '-pthread']` which is wrong. In that case the variable must be escaped so `pkg-config --cflags` would print `-I/home/Xavier\ Claessens/include -pthread` which gets split as `['-I/home/Xavier Claessens/include', '-pthread']`.

The common case is to have custom variables for path to a binary, like `myapp=/home/Xavier Claessens/app`, in that case it's not a problem because even if someone is expecting that to be escaped, they would have code that replace `\ ` back to ` ` which will be noop now.

So this is not backward compatible for projects who use custom variable inside `cflags:` or `libs:` fields like above... I hope...

nirbheek commented:
We could add a new kwarg that specifies whether the value is a path or not?

tp-m commented:
> We could add a new kwarg that specifies whether the value is a path or not?

or a kwarg that provides a list of variables that should not be escaped?

xclaesse commented:
> or a kwarg that provides a list of variables that should not be escaped?

Done that: `unescaped_variables` and `unescaped_uninstalled_variables`.

jpakkane commented:
LGTM, test failure is unrelated.
